File: php/triangle/triangle.php
<?php

class Triangle
{
    public function __construct($a, $b, $c)
    {
        $sides = [$a, $b, $c];

        if (!array_sum($sides)) {
            throw new Exception("No size is illegal");
        }

        sort($sides);
        if (($sides[0] + $sides[1]) < $sides[2]) {
            throw new Exception("Triangle inequality mof'");
        }

        $same = array_unique($sides);

        $kinds = [
            1 => "equilateral",
            2 => "isosceles",
            3 => "scalene",
        ];

        $this->kind = $kinds[count($same)];
    }

    public function kind()
    {
        return $this->kind;
    }
}

if (!debug_backtrace()) {
    new Triangle(1, 3, 1);
}

File: python/triangle/triangle.py
from typing import List

def diff_sides(sides: List[int]) -> int:
    sides = sorted(sides)

    def is_invalid():
        return (
            (sides[0] + sides[1] < sides[2])  # impossible triangle
            or (sum(sides) == 0)  # is it a point ?
        )

    return (
        0
        if is_invalid()
        else len(set(sides))  # qty of differtents sides length
    )

def equilateral(sides: List[int]) -> bool:
    return diff_sides(sides) == 1


def isosceles(sides: List[int]) -> bool:
    return diff_sides(sides) in [1, 2]  # equilateral is isoceles as well


def scalene(sides: List[int]) -> bool:
    return diff_sides(sides) == 3
